### PR TITLE
Assorted minor nits in libcontainer

### PR DESCRIPTION
**Pull request by kolyshkin**
Please see the individual commits for details.

### CODE REVIEW

File: libcontainer/cgroups/systemd/unified_hierarchy.go [hunk 1]
```diff
@@ -211,15 +212,8 @@ func createCgroupsv2Path(path string) (Err error) {
...
-	for i, c := range strings.Split(strings.TrimSpace(string(content)), " ") {
-		if i == 0 {
-			res = fmt.Sprintf("+%s", c)
-		} else {
-			res = res + fmt.Sprintf(" +%s", c)
-		}
-	}
-	resByte := []byte(res)
+	ctrs := bytes.Fields(content)
+	res := append([]byte("+"), bytes.Join(ctrs, []byte(" +"))...)
```
Comment (kolyshkin) [inline libcontainer/cgroups/systemd/unified_hierarchy.go]: @giuseppe PTAL

File: libcontainer/cgroups/utils.go [hunk 2]
```diff
@@ -524,14 +524,13 @@ func GetAllPids(path string) ([]int, error) {
...
-			return nil
-		}
 		if iErr != nil {
 			return iErr
 		}
-		cPids, err := readProcsFile(dir)
+		if info.IsDir() || info.Name() != CgroupProcesses {
+			return nil
+		}
+		cPids, err := readProcsFile(p)
```
Comment (kolyshkin) [inline libcontainer/cgroups/utils.go]: Previous change to this code was in https://github.com/opencontainers/runc/pull/332

File: libcontainer/cgroups/utils.go [hunk 3]
```diff
@@ -514,8 +514,8 @@ func getHugePageSizeFromFilenames(fileNames []string) ([]string, error) {
 }
 
 // GetPids returns all pids, that were added to cgroup at path.
-func GetPids(path string) ([]int, error) {
-	return readProcsFile(path)
+func GetPids(dir string) ([]int, error) {
```
Comment (AkihiroSuda) [inline libcontainer/cgroups/utils.go]: breaks compatibility if there is non-runc consumer of this function

Comment (kolyshkin) [reply libcontainer/cgroups/utils.go]: Nope (I just changed the name of the parameter to be clear it's a directory, otherwise it's doing the same thing).

**Comment by AkihiroSuda** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**